packages/server-core/src/recording/recording/recording.class.ts

```diff
@@ -23,99 +23,90 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
+export class RecordingService<T = RecordingType, ServiceParams extends Params = RecordingParams> extends KnexAdapter<
+  RecordingType,
+  RecordingData,
+  RecordingParams,
+  RecordingPatch
+> {
   app: Application
-  docs: any
-  constructor(options: Partial<SequelizeServiceOptions>, app: Application) {
+
+  constructor(options: KnexAdapterOptions, app: Application) {
     super(options)
     this.app = app
   }
 
-  async get(id: RecordingID, params?: any): Promise<T> {
-    // get resources with associated URLs
-    // TODO: move resources population to resolvers once this service is migrated to feathers 5
-    const resources = await this.app.service(recordingResourcePath).find({
-      query: {
-        recordingId: id
-      },
-      paginate: false
-    })
-
-    const result = (await super.get(id)) as RecordingDataType
-
-    result.resources = resources.map((resource) => resource.staticResource)
-
-    return result as T
+  async get(id: RecordingID, params?: RecordingParams) {
+    return await super._get(id, params)
   }
 
-  async find(params?: UserParams): Promise<Paginated<T>> {
+  async find(
+    params?: RecordingParams & {
+      paginate?: PaginationOptions | false
+    }
+  ) {
+    let isAdmin = false
     if (params && params.user && params.query) {
       const admin = await checkScope(params.user, this.app, 'admin', 'admin')
       if (admin && params.query.action === 'admin') {
         delete params.query.action
         // show admin page results only if user is admin and query.action explicitly is admin (indicates admin panel)
-        const recordings = (await super.find({ ...params })) as Paginated<RecordingDataType>
-
-        // TODO: Move this to resolvers as recordings service is moved to feathers 5
-        const recordingUsers = (await this.app.service(userPath)._find({
-          query: {
-            id: {
-              $in: recordings.data.map((item) => item.userId)
-            }
-          },
-          paginate: false
-        })) as any as UserType[]
-
-        for (const recording of recordings.data) {
-          const user = recordingUsers.find((item) => item.id === recording.userId)
-          recording.userName = user ? user.name : ''
-        }
-
-        // TODO: Remove as any once this file is migrated to feathers 5.
-        return recordings as any
+        isAdmin = true
       }
     }
 
-    return (await super.find({
-      query: {
-        userId: params?.user!.id
-      }
-    })) as Paginated<T>
+    const paramsWithoutExtras = {
+      ...params,
+      // Explicitly cloned sort object because otherwise it was affecting default params object as well.
+      query: params?.query ? JSON.parse(JSON.stringify(params?.query)) : {}
+    }
+
+    // Remove recording username sort
+    if (paramsWithoutExtras.query?.$sort && paramsWithoutExtras.query?.$sort['user']) {
+      delete paramsWithoutExtras.query.$sort['user']
+    }
+
+    if (isAdmin) {
```

Comment:
If it was admin then just manipulate the query parameter as per requirement instead of having multiple super._find calls.